lua/entities/hl_camera/init.lua

```diff
@@ -128,13 +129,66 @@ function ENT:SetPlayerKey(ply, key, toggle)
 	self:UpdatePlayer(ply)
 end
 
+function ENT:GhostCreate()
+	if IsValid(self.ACGhost) then
+		return
+	end
+	local ghost = ents.Create("hl_camera_ghost")
+
+	ghost:Spawn()
+
+	ghost:SetParent(self)
+	ghost:SetLocalPos(self:GetViewOffset())
+	ghost:SetLocalAngles(Angle(0, 0, self:GetRoll()))
+
+	return ghost
+end
+
+function ENT:UpdateGhostVar(varchanged, oldvalue, newvalue)
+	if oldvalue == newvalue then return end
+
+	if varchanged == "EnableGhost" then
+		self:SetGhost(newvalue)
+	elseif varchanged == "ViewOffset" then
+		self:SetGhostOffset(newvalue, self:GetRoll(), self:GetEnableGhost())
+	elseif varchanged == "Roll" then
+		self:SetGhostOffset(self:GetViewOffset(), newvalue, self:GetEnableGhost())
+	end
+
+end
+
+function ENT:SetGhost(enabled)
```

Comment:
Rename this to `ENT:SwitchGhost(on)`, similar to `ENT:SwitchProjection(on)` in projection.lua. Similarly, rename the other functions to `ENT:UpdateGhost*` instead of `ENT:SetGhost*`.